### ISSUE TITLE
[Accessibility] Narrator cannot announce the indeterminate state of DataGridViewCheckBoxCell correctly

### ISSUE THREAD

**Issue by v-elnovikova** [dotnet/winforms#9256]
### .NET version

.NET SDK 8.0.100-preview.4.23260.5

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

When set DataGridViewCheckBoxCell  value to Indeterminate, Narrator cannot announce that state is Indeterminate, but "Unchecked".

![DataGridViewCeckBoxCell - indeterminate announced as unchecked in 8 0](https://github.com/dotnet/winforms/assets/113603457/bb8e6551-8967-43ab-994e-022ca9b914ad)

![DataGridViewCeckBoxCell - indeterminate is Off in 8 0](https://github.com/dotnet/winforms/assets/113603457/9d124cd1-5a86-49d4-994d-46b933d0ec81)

### Steps to reproduce

1. Create a Winforms .Net Core application with DataGridView.
2. Add a DataGridViewCheckBoxColumn to DataGridView.
3. Build and run application.
4. Launch Narrator to focus on a grid check box cell.

Sample app:  [WinFormsApp1.zip](https://github.com/dotnet/winforms/files/11688232/WinFormsApp1.zip)

**Comment by Amy-Li03** [comment]
Verified this on latest .NET 8.0.100-preview.7.23362.29, issue was fixed, Narrator can announce the indeterminate state of DataGridViewCheckBoxCell.
![image](https://github.com/dotnet/winforms/assets/62929087/de9465db-4c27-4479-a724-5f163658667b)
![image](https://github.com/dotnet/winforms/assets/62929087/aa58b8a5-5c08-48dc-b4de-9b0f8380fa05)

**Comment by Olina-Zhang** [comment]
Verified with .NET SDK 8.0.100-preview.7.23376.3 test pass build, this issue was fixed with above same verification result.
